packages/cli/src/commands/link/link.js

```diff
@@ -32,60 +33,57 @@ type FlagsType = {
  *             only that package is processed.
  */
 function link([rawPackageName]: Array<string>, ctx: ContextT, opts: FlagsType) {
-  let platforms;
-  let project;
-  try {
-    platforms = getPlatforms(ctx.root);
-    logger.debug(
-      'Available platforms: ' +
-        `${Object.getOwnPropertyNames(platforms)
-          .map(platform => getPlatformName(platform))
-          .join(', ')}`,
-    );
-    if (opts.platforms) {
-      platforms = pick(platforms, opts.platforms);
-    }
-    logger.debug(
-      'Targeted platforms: ' +
-        `${Object.getOwnPropertyNames(platforms)
-          .map(platform => getPlatformName(platform))
-          .join(', ')}`,
-    );
-    project = getProjectConfig(ctx, platforms);
-  } catch (err) {
-    logger.error(
-      'No package found. Are you sure this is a React Native project?',
-    );
-    return Promise.reject(err);
+  let platforms = ctx.platforms;
+  let project = ctx.project;
+
+  if (opts.platforms) {
+    platforms = pick(platforms, opts.platforms);
+    logger.debug('Skipping selected platforms');
   }
 
+  logger.debug(
+    'Available platforms: ' +
+      `${Object.keys(platforms)
+        .map(getPlatformName)
+        .join(', ')}`,
+  );
+
   if (rawPackageName === undefined) {
     logger.debug(
       'No package name provided, will attemp to link all possible packages.',
     );
-    return linkAll(ctx, platforms, project);
+    return linkAll(ctx, platforms);
   }
 
-  logger.debug(`Package to link: ${rawPackageName}`);
-
   // Trim the version / tag out of the package name (eg. package@latest)
   const packageName = rawPackageName.replace(/^(.+?)(@.+?)$/gi, '$1');
 
-  const dependencyConfig = getDependencyConfig(ctx, platforms, packageName);
+  if (!Object.keys(ctx.dependencies).includes(packageName)) {
+    throw new Error(dedent`
+      Unknown dependency. 
```

Comment:
@grabbou I get this error while linking `appcenter`. Is this because you only try to detect `react-native-` etc modules as deps? We should allow all of them I guess

Reply:
Yes. What is the command you are running? I haven't changed the discovery behaviour. I could look at all `package.json` dependencies, but that would be a change to `link` command too.

Reply:
This is the behaviour. We might consider changing that at some point, but it's out of scope for this PR.